modules/irsa/main.tf

```diff
@@ -38,8 +38,9 @@ resource "aws_iam_role" "irsa" {
         },
         "Action" : "sts:AssumeRoleWithWebIdentity",
         "Condition" : {
-          "StringLike" : {
-            "${var.addon_context.eks_oidc_issuer_url}:sub" : "system:serviceaccount:${var.kubernetes_namespace}:${var.kubernetes_service_account}"
+          "StringEquals" : {
+            "${var.addon_context.eks_oidc_issuer_url}:sub" : "system:serviceaccount:${var.kubernetes_namespace}:${var.kubernetes_service_account}",
+            "${var.addon_context.eks_oidc_issuer_url}:aud" : "sts.amazonaws.com"
```

Comment:
@satveerkhurpa how did you arrive at this change? I want to understand possible ramifications of this.

cc: @vara-bonthu, @Zvikan, @bryantbiggs

Reply:
this is similar to what we have upstream https://github.com/terraform-aws-modules/terraform-aws-iam/blob/master/modules/iam-role-for-service-accounts-eks/main.tf#L16-L26

Reply:
Yes, I verified the upstream codebase and updated accordingly.